File: main/parsing_result_view/get_data_for_result.py

from main.data_base.query_to_parsing import select_filtered_results
import json

# ├── filteredQuery: {}
# │   ├──  domain_list: []
# │   │   │   ├──  domain_id: str
# │   ├──  start_date: str
# │   ├──  end_date: str
# │   ├──  only_last: bool
# │   ├──  get_checked: bool
# │   ├──  get_unchecked: bool
# пока не отправляется
# │   ├──  get_with_out_price: bool

def get_results_list(filteredQuery):
    results_list = []
    # print('Создаем запрос')
    start_date = filteredQuery['start_date'].split("-")
    filteredQuery['start_date'] = "/".join([start_date[2], start_date[1], start_date[0]])
    end_date = filteredQuery['end_date'].split("-")
    filteredQuery['end_date'] = "/".join([end_date[2], end_date[1], end_date[0]])

    if len(filteredQuery['domain_list']) == 0:
        del filteredQuery['domain_list']
        # print(filteredQuery)

    res_list = select_filtered_results(filteredQuery)
    # print(res_list)
    for el in res_list:
        # print(el.id)
        results_list.append(el.id)

    return json.dumps(results_list)

File: main/static/js/result/getDataForResult.js
// пользователь формирует запрос для сервера
// ├── filteredQuery: {}
// │   ├──  domain_list: []
// │   │   │   ├──  domain_id: str
// │   ├──  start_date: str
// │   ├──  end_date: str
// │   ├──  only_last: bool
// │   ├──  get_checked: bool
// │   ├──  get_unchecked: bool
// пока не реализован
// │   ├──  get_with_out_price: bool

function readFilteredQuery(){
  filteredQuery = {}
  filteredQuery['domain_list'] = listDomainId
  filteredQuery['start_date'] = startDate.value
  filteredQuery['end_date'] = endDate.value
  filteredQuery['only_last'] = lastResult.checked
  filteredQuery['get_checked'] = onlyCheckedTrue.checked
  filteredQuery['get_unchecked'] = onlyCheckedFalse.checked
  // filteredQuery['get_with_out_price'] = withOutPrice.checked
  return filteredQuery
}

function getResultList() { // onclick в HTML ФОРМЕ
  let filteredQueryObject = readFilteredQuery()
  let filteredQuery = JSON.stringify(filteredQueryObject)
  console.log(filteredQuery)
  // получает список id результатов
  $.ajax({
    url: `/filtered_results`,
    contentType: 'application/json',
    type: 'POST',
    dataType : 'json',
    data: filteredQuery,
    success: function(response) {
      listResultId = response
      console.log(listResultId)
      fillListBlock()
      changeResult(listResultId[0])
    }
  })
}
